cpp/src/search/search.cu

```diff
@@ -100,13 +101,21 @@ std::unique_ptr<column> search_ordered(table_view const& t,
 
   // This utility will ensure all corresponding dictionary columns have matching keys.
   // It will return any new dictionary columns created as well as updated table_views.
-  auto matched  = dictionary::detail::match_dictionaries({t, values}, stream);
-  auto d_t      = table_device_view::create(matched.second.front(), stream);
-  auto d_values = table_device_view::create(matched.second.back(), stream);
+  auto matched = dictionary::detail::match_dictionaries({t, values}, stream);
+
+  // 0-table_view, 1-column_order, 2-null_precedence, 3-validity_columns
+  auto flattened_t =
```

Comment:
`*_t` usually implies a _type_. Using it as a variable name is confusing.

Reply:
How about `t_flattened` and `values_flattened?